json_encoder.go

```diff
@@ -182,19 +182,20 @@ func (enc *jsonEncoder) WriteEntry(sink io.Writer, msg string, lvl Level, t time
 		return errNilSink
 	}
 
-	final := enc.Clone().(*jsonEncoder)
-
-	// TODO: When we flatten the serialized messages, remove the extra truncate
-	// and copy below. This will change the shape of the serialized message, which
-	// will require updating many tests.
+	final := jsonPool.Get().(*jsonEncoder)
 	final.truncate()
 	final.bytes = append(final.bytes, '{')
-	final.messageF(msg).AddTo(final)
-	final.levelF(lvl).AddTo(final)
-	final.timeF(t).AddTo(final)
-	final.bytes = append(final.bytes, `,"fields":{`...)
-	final.bytes = append(final.bytes, enc.bytes...)
-	final.bytes = append(final.bytes, "}}\n"...)
+	enc.levelF(lvl).AddTo(final)
+	enc.timeF(t).AddTo(final)
+	enc.messageF(msg).AddTo(final)
+	if len(enc.bytes) > 0 {
+		if len(final.bytes) > 1 {
+			// All the formatters may have been no-ops.
+			final.bytes = append(final.bytes, ',')
+		}
+		final.bytes = append(final.bytes, enc.bytes...)
+	}
+	final.bytes = append(final.bytes, "}\n"...)
```

Comment:
Let's change this to:
```go
final.bytes = append(final.bytes, '}', '\n')
```

It looks like this is slightly faster,
https://github.com/prashantv/go-bench/blob/master/append_bytes_test.go
```
BenchmarkAppendString-8 500000000                3.72 ns/op            0 B/op          0 allocs/op
BenchmarkAppendBytes-8  2000000000               1.63 ns/op            0 B/op          0 allocs/op
```